docs/proposals/PROPOSAL_D.md

```diff
@@ -0,0 +1,480 @@
+      }
+    },
+    "edges": [
+      {
+        "source": "mpi",
+        "target": "mpi.version",
+        "relation": "contains"
+      },
+      {
+        "source": "mpi",
+        "target": "mpi.implementation",
+        "relation": "contains"
+      },
+      {
+        "source": "mpi",
+        "target": "mpi.portability",
+        "relation": "contains"
+      },
+      {
+        "source": "mpi.portability",
+        "target": "mpi.portability.optimization",
+        "relation": "contains"
+      },
+      {
+        "source": "mpi.portability",
+        "target": "mpi.portability.mode",
+        "relation": "contains"
+      },
+      {
+        "source": "os",
+        "target": "os.name",
+        "relation": "contains"
+      },
+      {
+        "source": "os",
+        "target": "os.release",
+        "relation": "contains"
+      },
+      {
+        "source": "os",
+        "target": "os.vendor",
+        "relation": "contains"
+      },
+      {
+        "source": "os",
+        "target": "os.version",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware",
+        "target": "hardware.gpu",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.enabled",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.driver",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu.driver",
+        "target": "hardware.gpu.driver.version",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.architecture",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.version",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu",
+        "target": "hardware.gpu.cuda",
+        "relation": "contains"
+      },
+      {
+        "source": "hardware.gpu.cuda",
+        "target": "hardware.gpu.cuda.version",
+        "relation": "contains"
+      },
+      {
+        "source": "communication",
+        "target": "communication.framework",
+        "relation": "contains"
+      }
+    ],
+    "metadata": {
+      "version": "0.0.0",
+      "source": "https://github.com/supercontainers/compspec"
+    }
+  }
+}
+```
+
+## Compatibility Artifact
+
+The compatibility artifact would include a subset of metadata attributes that are known to a schema, along with versions and links to the schemas used.
+
+### Example Compatibility Artifact
+
+From the example above we might derive the following compatibility artifact.
+
+```json
+{
+  "version": "0.0.0",
+  "kind": "CompatibilitySpec",
+  "metadata": {
+    "name": "lammps-prototype",
+    "jsonSchema": "https://raw.githubusercontent.com/supercontainers/compspec/main/supercontainers/compspec.json"
+  },
+  "compatibilities": [
```

Comment:
+1 for the `compatibilities` format. It solved my headache of overcomplicated spec where I mixed graphs with simple data representation.

Reply:
Nice! I'm glad it was helpful, it made sense to me to have this as a list to allow multiple groups, each with their own version, etc.